app/src/main/java/com/github/polypoly/app/ui/game/GameActivity.kt

```diff
@@ -25,79 +34,84 @@ import org.osmdroid.views.overlay.Marker
  */
 class GameActivity : ComponentActivity() {
 
-    // mock current Player
-    private val currentPlayer = Player(
-        user = User(
-            id = 4572,
-            name = "User test 1",
-            bio = "",
-            skin = Skin.default(),
-            stats = Stats(0, 0, 0, 0, 0),
-            trophiesWon = listOf(),
-            trophiesDisplay = mutableListOf(),
-        ),
-        balance = 420,
-        ownedLocations = listOf(),
-        roundLost = null,
-    )
-
-    // mock List of Players
-    private val players = listOf(
-        Player(
-            user = User(
-                id = 4573,
-                name = "User test 2",
-                bio = "",
-                skin = Skin.default(),
-                stats = Stats(0, 0, 0, 0, 0),
-                trophiesWon = listOf(),
-                trophiesDisplay = mutableListOf(),
-            ),
-            balance = 32,
-            ownedLocations = listOf(),
-            roundLost = null,
-        ), Player(
-            user = User(
-                id = 4574,
-                name = "User test 3",
-                bio = "",
-                skin = Skin.default(),
-                stats = Stats(0, 0, 0, 0, 0),
-                trophiesWon = listOf(),
-                trophiesDisplay = mutableListOf(),
-            ),
-            balance = 56,
-            ownedLocations = listOf(),
-            roundLost = null,
-        )
-    )
+    private val gameModel: GameViewModel by viewModels { GameViewModel.Factory }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContent {
+        setContent { GameActivityContent() }
+    }
+
+    @Composable
+    fun GameActivityContent() {
+        val player = gameModel.getPlayerData().observeAsState().value
+        val game = gameModel.getGameData().observeAsState().value
+        val gameTurn = gameModel.getRoundTurnData().observeAsState().value
+        val gameEnded = gameModel.getGameFinishedData().observeAsState().value
+
+        if (player != null && game != null && gameTurn != null && gameEnded != null) {
```

Comment:
What does this iteration of conditions define? Is it to check if a game is occurring ? Maybe for clarity, creating a method with an adequate name would provide easier understanding of the code. But I admit this is neat picking ;)

Reply:
Not exactly, these checks solve the fact that the value returned by `observeAsState().value` may be null. This is actually the recommended way to do it safely that I found in the official android doc.